### PR TITLE
Feature/security

### PR DESCRIPTION
**Pull request by KahKitZheng**
- Bcrypt password
- API protection, can only call API routes if you have a cookie
- Roles in edit user is now retrieved from database instead of hardcoded

### CODE REVIEW

File: client/src/views/EditUser.vue [hunk 1]
```diff
@@ -37,17 +37,16 @@ export default {
...
-      { id: "7", text: "Roosteraar" },
-      { id: "8", text: "Coordinator" }
+      // { id: "1", text: "Administrateur" },
+      // { id: "2", text: "Docent" },
+      // { id: "3", text: "Onderwijsprogrammacoordinator" },
+      // { id: "4", text: "Projectcoordinator" },
+      // { id: "5", text: "Modulecoordinator" },
+      // { id: "6", text: "Facilitator" },
+      // { id: "7", text: "Roosteraar" },
+      // { id: "8", text: "Coordinator" }
```
Comment (Aviumaa) [inline client/src/views/EditUser.vue]: ff comments wghalen

**Review by Aviumaa — APPROVED** [review]

File: client/src/views/EditUser.vue (hunk 1 above)
Comment (KahKitZheng) [reply client/src/views/EditUser.vue]: Fixed

File: client/src/views/EditUser.vue [hunk 2]
```diff
@@ -69,6 +58,14 @@ export default {
     }
   },
   mounted() {
+    axios.get("http://localhost:3000/api/roles/").then(response => {
+      for (let i = 0; i < response.data.length; i++) {
+        let role = response.data[i];
+        this.roles.push(role);
+      }
+      console.log(this.roles);
```
Comment (MarcoBergsma) [inline client/src/views/EditUser.vue]: Niet meer nodig denk ik?

Comment (KahKitZheng) [reply client/src/views/EditUser.vue]: Klopt, ik ga het weghalen

File: server/controllers/userController.js [hunk 3]
```diff
@@ -1,52 +1,95 @@
...
+  // decode token
+  if (token) {
+    // verifies secret and checks if the token is expired
+    jwt.verify(token, "secretkey", (err, decoded) => {
+      if (err) {
+        return res.json({ message: err });
+      } else {
+        // if everything is good, save to request for use in other routes
+        req.decoded = decoded;
+        console.log("token is gucci");
```
Comment (MarcoBergsma) [inline server/controllers/userController.js]: Pls haal dit weg.

File: server/controllers/userVacancyController.js [hunk 4]
```diff
@@ -1,31 +1,29 @@
...
-        res.status(200).json(userResponse)
-    });
-}
+  var vacancyId = req.params.vacancyId;
+  UserVacancy.findOne({
+    where: {
+      id: vacancyId
+    }
+  }).then(userResponse => {
+    console.log(req.param);
```
Comment (MarcoBergsma) [inline server/controllers/userVacancyController.js]: weg

**Review by MarcoBergsma — APPROVED** [review]

**Pull request merged**